dashboard/api_test.py

```diff
@@ -1827,18 +1827,40 @@ def test_program_financial_aid(self, mock_fin_aid_serialize, mock_info_course):
 @ddt.ddt
 class ExamSchedulableTests(MockedESTestCase):
     """Tests exam schedulable"""
+    @ddt.data(
+        (False, False, True),
+        (False, True, True),
+        (True, False, False),
+    )
+    @ddt.unpack
+    def test_is_exam_schedulable(self, is_past, is_future, can_schedule_exam):
+        """Test that is_exam_schedule is correct"""
+        now = now_in_utc()
+        exam_run = ExamRunFactory.create(
+            scheduling_past=is_past,
+            scheduling_future=is_future,
+            date_last_eligible=now.date()-timedelta(days=20) if is_past else now.date()+timedelta(days=20)
+        )
+        exam_auth = ExamAuthorizationFactory.create(exam_run=exam_run, course=exam_run.course)
+
+        assert api.is_exam_schedulable(exam_auth.user, exam_auth.course) is can_schedule_exam
+
     @ddt.data(
         (False, False),
-        (False, True),
-        (True, False),
+        (True, True),
     )
     @ddt.unpack
-    def test_is_exam_schedulable(self, is_past, is_future):
+    def test_is_exam_schedulable_future_exam_run(self, is_future, can_schedule_exam):
         """Test that is_exam_schedule is correct"""
-        exam_run = ExamRunFactory.create(scheduling_past=is_past, scheduling_future=is_future)
+        exam_run = ExamRunFactory.create(
+            eligibility_past=True,
+            eligibility_future=is_future,
```

Comment:
`eligibility_past` and `eligibility_future` set the same fields on the model, I think you could just set it once using either one of them.

Reply:
@annagav for `factory.Trait` we need to enable each flag to make it work, if it is set to false it want work. i,e `eligibility_future=False` then dates will not set to `eligibility_past`.
To run `eligibility_past` you must set it to True.

https://factoryboy.readthedocs.io/en/latest/introduction.html#altering-a-factory-s-behaviour-parameters-and-traits

Reply:
What is the meaning of the test for 
```eligibility_past=True, eligibility_future=True```
?